src/main/java/edu/kit/ipd/crowdcontrol/objectservice/rest/resources/ExperimentResource.java

```diff
@@ -318,6 +355,29 @@ private Experiment updateExperimentInfoDraftState(int id, Experiment experiment,
 
         experimentRecord.setDescription(Template.apply(descriptionRaw, placeholders));
 
+        try {
+            updateLinkedExperimentInfo(id, experiment, old, experimentRecord);
+        } catch (DataAccessException e) {
+            // restore old state
+            updateLinkedExperimentInfo(id, old, experiment, oldRecord);
+
+            throw new InternalServerErrorException("Updating the experiment info failed, restored old state.", e);
+        }
+
+        try {
+            //update the experiment itself
+            experimentOperations.updateExperiment(experimentRecord);
```

Comment:
Can you move those two try´s into one ? The exact fact if info or experiment failed can be seen in the bt. So we have duplicated code here.